src/services/socketOn.js

```diff
@@ -0,0 +1,24 @@
+// This file is used to handle the socket.on event
+
+module.exports = (io, socket) => {
+	socket.onAny((eventName, ...args) => {
+		console.log('>> eventName:', eventName)
+		console.log('>> args:', args)
+		const [message, userName, timeStamp, roomId] = args
+		// console.log('>> message, roomId, userName:', message, roomId, userName)
+		console.log('**************************************************')
+		if (eventName === 'join-room') {
+			socket.join(roomId)
+		}
+		io.to(roomId).emit(eventName, message, userName, timeStamp)
```

Comment:
suggestion: To make this generic for any type of event you can spread args, so that you don't need line 7 and 8.

```js
io.to(roomId).emit(...args)
```

Reply:
I would still need to have a line 7 with const [roomId, ....args], because I need roomId on line 13. So yes, it could take any number of args, but I don't want it to take any number. I want it to be exactly what is needed without additional arguments. So why make it generic?

Reply:
This function `socket.onAny` is generic by itself (`onAny`), so it implies it's called for anything in the app, not only rooms but anything really. That's why the args should be generic. But this is just a `suggestion`, up to you. As it is, it works just fine :)

Reply:
Okay, I get that. Actually, that's what I had originally thought of when choosing socket.onAny, but then narrowed to only have those specific arguments. I guess it really isn't necessary to have them specified EXCEPT roomId. Okay, I'll change that, too.

Reply:
Okay, that won't really work because the args have to be in a specific order. I would have to move roomId everywhere. I think I'll do that anyway so that it'll be easier to remove timestamp or add something else

Reply:
DONE

Reply:
After you check/learn about the "options object pattern" I mentioned in another comment, that refactor will become much easier ;)

Reply:
That's a good point. I might give that a try for the next PR